### DISCUSSION TITLE
Docker DNS

### DISCUSSION THREAD

**Discussion by justmytwospence** [qdm12/gluetun#511] [Q&A] [answered]
I have gluetun set up in one `docker-compose.yaml`, to which I have added an external network called `calibre-net`. In a second `docker-compose.yaml`, I have a Calibre docker container running on the `calibre-net` network with the `container_name` `calibre`. From the Gluetun container (or containers attached to it via `network_mode: "service:gluetun"`), I am unable to ping the host `calibre`, although I am able to ping the IP address directly. Is there a way to get DNS to resolve `calibre` with the way things are currently set up?

**Comment by qdm12** [comment, marked as answer]
See #281 

For now, you can disable DNS over TLS and enable `DNS_KEEP_NAMESERVER` from the top of my head. Although your DNS traffic is going to leak out of the VPN, so not recommended. I'm working on a fix, coding a DNS server from scratch (https://github.com/qdm12/dns/tree/v2.0.0-beta) but this is still in beta and likely to take me a few more weeks.

**Reply by justmytwospence** [reply]
Following - impressive response time!

**Reply by techie2000** [reply]
For anyone reading this, be aware it is `DNS_KEEP_NAMESERVER` rather than `DNS_KEEPNAMESERVER` and takes a value of OFF|ON

**Reply by qdm12** [reply]
@techie2000 Thanks I updated my comment. I also created https://github.com/qdm12/gluetun/issues/562 to log out any env variable that are not recognized by the program.
